src/constructor.io/models/common/Facet.cs

```diff
@@ -0,0 +1,251 @@
+        /// </summary>
+        public string DisplayName { get; set; }
+
+        /// <summary>
+        /// Default criterion to sort this facet group. Overridden by `Position` attribute on facet options.
+        /// Defaults to `relevance`.
+        /// </summary>
+        public FacetSortOrderBy? SortOrder { get; set; }
+
+        /// <summary>
+        /// Sort direction for `SortOrderBy`.
+        /// Default value is `false` for SortOrderBy = `value` and `true` otherwise.
+        /// </summary>
+        public bool? SortDescending { get; set; }
+
+        /// <summary>
+        /// Type of facet group.
+        /// </summary>
+        public FacetType Type { get; set; }
+
+        /// <summary>
+        /// Specifies whether the range buckets are determined dynamically (based on the values of matching facet options) or statically.
+        /// Required if FacetType = `range` & RangeFormat = `options`.
+        /// </summary>
+        public FacetRangeType? RangeType { get; set; }
+
+        /// <summary>
+        /// Format of range facets. Determines whether the facet is configured to display as a slider (in which case the search endpoint will return only min & max values) or as a list of buckets.
+        /// Required if FaceType = `range`.
+        /// </summary>
+        public FacetRangeFormat? RangeFormat { get; set; }
+
+        /// <summary>
+        /// Used to create inclusive buckets.
+        /// </summary>
+        [JsonProperty(NullValueHandling = NullValueHandling.Include)]
+        public FacetRangeInclusive? RangeInclusive { get; set; }
+
+        // Used by Json.Net to determine whether or not to serialize RangeInclusive.
+        public bool ShouldSerializeRangeInclusive()
+        {
+            if (RangeInclusive == null) return false;
+            if (RangeInclusive == FacetRangeInclusive.Null) RangeInclusive = null;
+            return true;
+        }
+
+        /// <summary>
+        /// Specifies the size of generated range buckets. Defaults to null.
+        /// Required if FacetType = `range` & RangeLimits = null.
+        /// </summary>
+        public float? BucketSize { get; set; }
+
+        /// <summary>
+        /// Defines the cut-off points for generating static range buckets. Expects list of sorted numbers (like [10, 25, 40]).
+        /// Required if FacetType = `range` & BucketSize = null.
+        /// </summary>
+        public List<int> RangeLimits { get; set; }
+
+        /// <summary>
+        /// Specifies filter behavior given multiple filters on the same facet (e.g: color: yellow & blue) are selected.
+        /// </summary>
+        public FacetMatchType? MatchType { get; set; }
+
+        /// <summary>
+        /// Used to slot facet groups to fixed positions.
+        /// </summary>
+        [JsonProperty(NullValueHandling = NullValueHandling.Include)]
+        public int? Position { get; set; }
+        public bool ShouldSerializePosition()
```

Comment:
what does this function do?

Reply:
ah, this was the thing we discussed in the huddle a couple of days ago.

So because C# doesn't have a `undefined` equivalent, there's no way to tell if the SDK consumer has set `Position` to `null` or if it was initialized to null. This becomes relevant in methods like [Partial Facet Config Update](https://docs.constructor.io/rest_api/facets/#update-facet-configs-partial) which allow for you to modify `Position` to null to remove previously configured slotting.

This method tells JsonConvert to:
-  serialize the `Position` property as `null` if it was set to `-1` by the SDK consumer
- not serialize if it currently has the initialized value `null` (because this would result in unintentional modification)
- serialize any other value

Essentially if you wish to intentionally set `Position = null`, you'd have to set it to `-1` using this SDK. Not ideal, but the alternative would be to request a Hashtable object where they manually add `position: null`

Reply:
Let's see if we can do the reverse and default to the value to `-1`. That way consumers can specify `null` as an actual value like they would expect it from looking at our API docs

Reply:
Seems like we can! I've pushed the changes up :D This would definitely be a more intuitive way to consume the library. +1